### PR TITLE
ci: use alias latest and minimum to represent xcode cmd tool version

### PR DESCRIPTION
**Pull request by 5d**
## Description

- add alias `latest` and `minimum` to represent xcode cmd tool version
  - `latest` is xcode 14.3.1
  - `minimum` is xcode 14.0.1

## General Checklist

- [ ] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [X] All unit tests pass
- [x] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [X] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/integ_test_logging.yml [hunk 1]
```diff
@@ -39,7 +39,6 @@ jobs:
       scheme: ${{ matrix.platform == 'watchOS' && 'AWSCloudWatchLoggingPluginIntegrationTestsWatch' || 'AWSCloudWatchLoggingPluginIntegrationTests' }}
       platform: ${{ matrix.platform }}
       project_path: ./AmplifyPlugins/Logging/Tests/AWSCloudWatchLoggingPluginHostApp
-      xcode_version: ${{ matrix.platform == 'watchOS' && '15.0' || '14.3' }}
```
Comment (ruisebas) [inline .github/workflows/integ_test_logging.yml:42]: These tests used to run on Xcode 15.0 for watchOS, but now they will use 14.3.
Have you checked it still works?

File: .github/composite_actions/get_platform_parameters/action.yml [hunk 2]
```diff
@@ -5,62 +5,99 @@ inputs:
     required: true
     type: string
   xcode_version:
-    description: 'The version of Xcode. Valid values are 14.3 and 15.0'
-    required: true
+    description: "The version of Xcode. Available aliases are 'latest' and 'minimum'"
+    default: 'latest'
+    type: string
+  destination:
```
Comment (ruisebas) [inline .github/composite_actions/get_platform_parameters/action.yml:11]: Is this new `destination` input meant to override the one that is automatically determined by the Xcode version?

When is it used or needed?

File: .github/composite_actions/get_platform_parameters/action.yml [hunk 3]
```diff
@@ -5,62 +5,99 @@ inputs:
     required: true
     type: string
   xcode_version:
-    description: 'The version of Xcode. Valid values are 14.3 and 15.0'
-    required: true
+    description: "The version of Xcode. Available aliases are 'latest' and 'minimum'"
```
Comment (ruisebas) [inline .github/composite_actions/get_platform_parameters/action.yml:8]: Why not just add the new aliases but still support passing an explicit version? 

Because now `"latest`" is not really ***the*** latest (i.e. 15.0), so it's kinda misleading to call it like that.
If we have `latest = 15.0` set, but we explicitly pass `14.3` in our tests invocations, it's clear we need to investigate why we cannot upgrade.

So basically what I'm saying is keeping `14.0.1`,  `14.3` and `15.0` as allowed value, but also supporting `latest` which we consider to be `15.0` and `minimum` which we consider to be `14.0.1`.

Then you can just use `latest` in tests that support it, but set `14.3` in the ones that don't.

**Review by ruisebas — CHANGES_REQUESTED** [review]
Left some comments/suggestions.

Also, have you run all the integration tests suits to check they also pass with the new Xcode version?

File: .github/composite_actions/get_platform_parameters/action.yml (hunk 2 above)
Comment (5d) [reply .github/composite_actions/get_platform_parameters/action.yml:11]: Yes, it is used to define the exact destiation. In case of `latest` or `minimum`, you can use the default empty string value and the script will try to use the predefined destination mappings.
This input option becomes essential when testing a particular Xcode version that is neither `latest` nor `minimum`. For example, when testing a beta Xcode version and specifying the beta SDK version in the `destination` input.

File: .github/composite_actions/get_platform_parameters/action.yml (hunk 3 above)
Comment (5d) [reply .github/composite_actions/get_platform_parameters/action.yml:8]: It is possible to pass a specific version. The alias available is only `latest` and `minimum`. You can still pass `15.1` as `xcode_version` input. However, you will also need to specify the `destination` for it.

File: .github/workflows/integ_test_logging.yml (hunk 1 above)
Comment (5d) [reply .github/workflows/integ_test_logging.yml:42]: good catch, will update it.

File: .github/composite_actions/get_platform_parameters/action.yml (hunk 3 above)
Comment (ruisebas) [reply .github/composite_actions/get_platform_parameters/action.yml:8]: Aah I see, thanks for clarifying that :) 

If setting an explicit destination is indeed an "edge" case (i.e. when testing new beta versions, or for a reduced number of tests), then that's fine. But if we have to write the whole `destination` in many files, then this action becomes irrelevant and exactly what it tried to replace :P

tl;dr: this is fine as long as we don't have to set an explicit xcode version in more than 1 workflow, otherwise we should add the mapped destination for that xcode version and save ourselves one long and messy input.

Comment (5d) [reply .github/composite_actions/get_platform_parameters/action.yml:8]: yeah, if that's the case, we might need to consider adding another alias `beta` and the destination mapping in the script.

**Review by ruisebas — APPROVED** [review]
